### PR TITLE
Update links and examples

### PR DESCRIPTION
**Pull request by geografa**
Starting this as a larger cleanup of docs and the API examples.

### CODE REVIEW

**Comment by andrewharvey** [comment]
Isn't API.md auto generated by https://github.com/mapbox/mapbox-sdk-js/blob/master/package.json#L13, so it's better to fix the source and then rerun `npm run docs`.

**Comment by andrewharvey** [comment]
See also #146 and https://github.com/mapbox/mapbox-sdk-js/commit/d10eed9cfe4b09b62783937725eddbeeaeb8fe2c

**Comment by mollymerp** [comment]
These tests are flappy! I restarted and seems to be passing now.

**Comment by geografa** [comment]
@scothis @mollymerp I was going to update services and change distance.js to matrix.js https://github.com/mapbox/mapbox-sdk-js/blob/master/lib/services/distance.js

Ok to do this here or should I make a new PR?

File: README.md [hunk 1]
```diff
@@ -31,14 +31,12 @@ Generally Available
 
 Contact help@mapbox.com for information
 
-* [Distance](https://www.mapbox.com/developers/api/distance/)
-  * Travel-time tables between up to 100 points
+* [Matrix](https://www.mapbox.com/api-documentation/#matrix)
```
Comment (mollymerp) [inline README.md]: I think this change is jumping the gun on #176 – once you revert this I think this PR will be good to go.

Comment (geografa) [inline README.md]: #176 has been merged into this so we should be gtg here

**Pull request merged**